Packages/Tracking Preview/PhysicsHands/Runtime/UI/Scripts/PhysicsButton.cs

```diff
@@ -0,0 +1,506 @@
+            OnUncontact += _OnUncontact.Invoke;
+            OnPressedAmountChanged += _OnPressedAmountChanged.Invoke;
+            OnToggle += _OnToggle.Invoke;
+        }
+
+        protected void FixedUpdate()
+        {
+            if (_provider != null)
+            {
+                ProcessPhysicsEvents();
+            }
+        }
+
+        private void ProcessPhysicsEvents()
+        {
+            if (_provider.IsObjectHovered(_buttonElement.Rigid))
+            {
+                if (!_hovered)
+                {
+                    _hovered = true;
+                    OnHover?.Invoke();
+                }
+
+                if (_provider.GetObjectState(_buttonElement.Rigid, out var state))
+                {
+                    switch (state)
+                    {
+                        case PhysicsGraspHelper.State.Hover:
+                            _contactReleaseTime = Mathf.Clamp(--_contactReleaseTime, 0, 100);
+                            if (_contact && _contactReleaseTime == 0)
+                            {
+                                _contact = false;
+                                OnUncontact?.Invoke();
+                            }
+                            break;
+                        case PhysicsGraspHelper.State.Contact:
+                        case PhysicsGraspHelper.State.Grasp:
+                            if (!_contact)
+                            {
+                                _contact = true;
+                                OnContact?.Invoke();
+                                // Small wait to reduce erroneous uncontact events
+                                _contactReleaseTime = 10;
+                            }
+                            // Ensure that we only allow hands to press the button
+                            if (_handsOnly)
+                            {
+                                PressValidation();
+                            }
+                            break;
+                    }
+                }
+            }
+            else
+            {
+                if (_hovered)
+                {
+                    _hovered = false;
+                    OnUnhover?.Invoke();
+                }
+                if (_contact)
+                {
+                    _contact = false;
+                    OnUncontact?.Invoke();
+                }
+            }
+
+            // If we want all physics objects to press it then go wild
+            if (!_handsOnly)
+            {
+                PressValidation();
+            }
+
+            if (_isPressed && _buttonElement.transform.localPosition.y >= buttonHeightLimit * 0.05f)
+            {
+                _isPressed = false;
+                _unpressedThisFrame = true;
+                OnUnpress?.Invoke();
+            }
+
+            pressedAmount = Mathf.InverseLerp(buttonHeightLimit * 0.99f, buttonHeightLimit * 0.01f, _buttonElement.transform.localPosition.y);
```

Comment:
What are these magic numbers about?

Reply:
Same principle as above comment but scales the pressed amount between the limits.

Reply:
Could we change the values to named constants?